Tactics/Reflect/MemoryEffects.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Alex Keizer, Siddharth Bhat
+-/
+
+import Arm.State
+import Tactics.Common
+import Tactics.Attr
+import Tactics.Simp
+
+import Std.Data.HashMap
+
+open Lean Meta
+
+section MetaMHelpers
+/-- Create a trace node in trace class (i.e. `set_option traceClass true`),
+with header `header`, whose default collapsed state is `collapsed`. -/
+def withTraceNode' (header : MessageData) (k : MetaM α)
+    (collapsed : Bool := true)
+    (traceClass : Name := `Tactic.sym) : MetaM α :=
+  Lean.withTraceNode traceClass (fun _ => return header) k (collapsed := collapsed)
+
+/-- Create a trace note that folds `header` with `(NOTE: can be large)`,
+and prints `msg` under such a trace node.
+-/
+def traceLargeMsg (header : MessageData) (msg : MessageData) : MetaM Unit :=
+    withTraceNode' m!"{header} (NOTE: can be large)" do
+      trace[Tactic.sym] msg
```

Comment:
Should these go to some more common file? I'm also not a huge fan of priming names, how about `SymM.withTraceNode` or some other similar namespace?